File: api/src/Monument.ts
export interface Monument {
    readonly datasetid : string,
    readonly recordid : string,
    readonly fields : {
        readonly insee : string,
        readonly objectid_1 : string,
        readonly lien_merim : string,
        readonly objectid : string,
        dep : string,
        type_archi : string,
        nomcom : string,
        geo_point_2d : number[],
        st_lengthshape : string,
        geo_shape : {
            type : string,
            //coordinates, c'est un tableau contenant d'autres tableaux de nombres
        }
        protection : string,
        immeuble : string,
        date_prot : string, //maybe Date
        st_areashape : string,
        ref_merim : string,
        type_prot : string,
    }
    geometry : {
        type : string,
        coordinates : number[],
    }
    record_timestamp : string, //maybe Date
}

File: android/app/src/main/java/com/ismin/android/Monument.kt
package com.ismin.android

import java.io.Serializable
import java.util.*

data class Monument (val immeuble : String,//nom monument
                     val dep : String,//nom département
                     val nomcom : String,//code postal commune
                     val datasetid : String,
                     val recordid : String,
                     val insee : String,
                     val objectid_1 : String,
                     val lien_merim : String,
                     val objectid : String,
                     val type_archi : String,
                     val geo_point_2d : FloatArray,//coordonnés
                     val st_lengthshape : String,
                     val protection : String,
                     val date_prot : Date,
                     val st_areashape : String,
                     val ref_merim : String,
                     val type_prot : String,
                     val type : String,
                     val coordinates : FloatArray,
                     val record_timestamp : Date,
                     val favorite : Boolean,):Serializable {}
